.github/workflows/main.yml

```diff
@@ -1,72 +1,145 @@
+            com.arkhn.image.created=${{ steps.metadata.outputs.created }}
+            com.arkhn.image.revision=${{ github.sha }}
+            com.arkhn.image.licenses=${{ github.event.repository.license.spdx_id }}
+      - uses: actions/upload-artifact@v2
+        with:
+          name: image
+          path: image
+
+  tests:
     needs: build
-    if: github.ref == 'refs/heads/master'
+    runs-on: ubuntu-18.04
+
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: image
+      - run: docker load -i image
+      - run: |
+          IMAGE=$(echo ${{ needs.build.outputs.tags }} | cut -f1 -d,)
+          docker run --entrypoint=pytest ${IMAGE}
+
+  publish-docker:
+    needs: [tests, build]
+    runs-on: ubuntu-18.04
+
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: image
+      - run: docker load -i image
+      - uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_LOGIN }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - run: |
+          TAGS=${{ needs.build.outputs.tags }}
+
+          for tag in ${TAGS//,/ }; do
+            docker push "${tag}"
+          done
+
+  publish-pypi:
+    needs: [tests, build]
+    runs-on: ubuntu-18.04
+    if: github.event_name == 'push' && github.event.base_ref == 'refs/heads/master' && startsWith(github.ref, 'refs/tags')
+
     steps:
       - uses: actions/checkout@v1
-      - name: Set up Python
-        uses: actions/setup-python@v1
+      - uses: actions/setup-python@v1
         with:
           python-version: "3.7"
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
+          pip install -r requirements.txt
           pip install -r requirements-dev.txt
-      - name: Build and publish
+      - name: Publish package to pypi
         env:
           TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
           TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
         run: |
-          python setup.py sdist bdist_wheel
-          ls -l dist/*.tar.gz
-          twine upload dist/*.tar.gz
-      - name: Build Docker image
-        run: docker build -t $IMAGE:$TAG .
-      - name: Publish to the docker hub
-        run: |
-          echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u ${{ secrets.DOCKER_LOGIN }} --password-stdin
-          docker push $IMAGE:$TAG
-      - name: Publish to Github Packages
-        run: |
-          echo "${{ secrets.GITHUB_TOKEN }}" | docker login $GITHUB_REGISTRY -u ${{ github.actor }} --password-stdin
-          docker tag $IMAGE:$TAG $GITHUB_REGISTRY/${{ github.repository }}/cleaning-scripts:$TAG
-          docker push $GITHUB_REGISTRY/${{ github.repository }}/cleaning-scripts:$TAG
+          VERSION="${GITHUB_REF#refs/tags/}"
+          # Does the tag follow numeric versioning ?
+          if [[ "${VERSION}" =~ ^v[0-9]{1,3}(\.[0-9]{1,3}(\.[0-9]{1,3})?)?$ ]]; then
+            VERSION="${VERSION#v}"
```

Comment:
Does this set the VERSION env var that is used in setup.py?

Reply:
yes it did, but I reverted that change in order to keep using the version manually specified in setup.py to publish on pypi.